cmd/backup/main.go

```diff
@@ -93,17 +96,66 @@ type script struct {
 	webdavClient *gowebdav.Client
 	logger       *logrus.Logger
 	sender       *router.ServiceRouter
+	template     *template.Template
 	hooks        []hook
 	hookLevel    hookLevel
 
-	start  time.Time
-	file   string
-	output *bytes.Buffer
+	file  string
+	stats *Stats
 
-	c *config
+	c *Config
 }
 
-type config struct {
+// ContainersStats stats about the docker containers
+type ContainersStats struct {
+	All        int
+	ToStop     int
+	Stopped    int
+	StopErrors int
+}
+
+// BackupFileStats stats about the created backup file
+type BackupFileStats struct {
+	Name      string
+	FullPath  string
+	SizeBytes int64
+	SizeBin   string
+	SizeDec   string
+}
+
+// ArchiveStats stats about the status of an archival directory
+type ArchiveStats struct {
+	Total       int
+	Pruned      int
+	PruneErrors int
+}
+
+// ArchivesStats stats about each possible archival location (Local, WebDAV, S3)
+type ArchivesStats struct {
+	Local  ArchiveStats
+	WebDAV ArchiveStats
+	S3     ArchiveStats
+}
+
+// Stats global stats regarding script execution
+type Stats struct {
+	StartTime  time.Time
+	EndTime    time.Time
+	TookTime   time.Duration
+	Output     *bytes.Buffer
+	Containers ContainersStats
+	BackupFile BackupFileStats
+	Archives   ArchivesStats
```

Comment:
`Archives` as a name makes sense, but somehow collides with the "tar archive" terminology. I can't really come up with a better name right now though. The docs use the term `Storage`, but `Storages` sounds _very_ odd.

In case something pops up in my mind soon I will let you know, if not I guess this is just fine.

Reply:
You're right. Also the word "storage" is mentioned a lot in the configuration reference, so it makes sense. I agree that `Storages` sounds a little odd, [but is apparently correct](https://www.wordhippo.com/what-is/the-plural-of/storage.html).

If you really don't like it, we could keep "storage", but avoid pluralizing and use something like `StorageTypes`, `StorageLocations`, `StorageList`, etc...

The only other somewhat similar word that I found is `Repository`, but I don't think it fits very well.

Reply:
I guess `Storages` is probably the most consistent term to use.

Reply:
Yeah I agree. `Storages` it is then?